src/components/pacbio/PacbioLibraryCreate.vue

```diff
@@ -70,8 +68,11 @@
             id="library-templatePrepKitBoxBarcode"
             v-model="library.template_prep_kit_box_barcode"
             type="text"
-            required
+            minlength="21"
+            maxlength="21"
             placeholder="Example: 012345678901234567890"
+            pattern="\d*"
+            inputmode="numeric"
```

Comment:
This is based on the data I've seen in production. It confines it to a 21 digit string. inputmode is used by phones and tablets to configure the keyboard that gets displayed.